src/processes/poisson.jl

```diff
@@ -19,61 +24,74 @@ Base.union(p₁::PoissonProcess{<:Real}, p₂::PoissonProcess{<:Function}) = Poi
 
 Base.union(p₁::PoissonProcess{<:Function}, p₂::PoissonProcess{<:Real}) = PoissonProcess(x -> p₁.λ(x) + p₂.λ)
 
+Base.union(p₁::PoissonProcess{<:Vector}, p₂::PoissonProcess{<:Vector}) = PoissonProcess(x -> p₁.λ + p₂.λ)
+
 ishomogeneous(p::PoissonProcess{<:Real}) = true
 ishomogeneous(p::PoissonProcess{<:Function}) = false
+ishomogeneous(p::PoissonProcess{<:Vector}) = false
 
-default_sampling_algorithm(::PoissonProcess, ::Geometry) = DiscretizedSampling()
-default_sampling_algorithm(::PoissonProcess{<:Real}, ::Box) = ProductSampling()
+default_sampling_algorithm(::PoissonProcess, ::Any) = DiscretizedSampling()
+default_sampling_algorithm(::PoissonProcess{<:Function}, ::Any) = ThinnedSampling()
```

Comment:
It seems that our default sampling algorithm is no longer a function of the geometry. Maybe we should simply erase the second argument?